### PR TITLE
Rewrite comment with kernel-doc style

### LINKED ISSUE #84: Write kernel-doc comments for queue.h
File `list.h` follows [kernel-doc comments](https://docs.kernel.org/doc-guide/kernel-doc.html), which allows the generation of API documentation extracted from the comments. However, file `queue.h` does not, and it would be nice to enforce the kernel-doc formatted documentation for functions.

Expected outcome:
1. Rewrite the comments in file `queue.h` which follows kernel-doc comments.
2. Simplify the comments in file `queue.c`: Only one-line summary should be kept as the comments. The programmers should read the generated documentation while modifying `queue.c`.
3. Move `q_release_element` from file `queue.c` to `queue.h` as a static inline function.

### PR DESCRIPTION
**Pull request by eecheng87**
Close #84

### CODE REVIEW

**Review by jserv — CHANGES_REQUESTED** [review]
Remove the wording "Attempt to" in file "queue.c"

File: queue.h [hunk 1]
```diff
@@ -22,35 +22,50 @@ typedef struct {
 
 /* Operations on queue */
 
-/* Create empty queue.
- * Return NULL if could not allocate space.
+/**
+ * q_new() - attempt to create empty queue
+ *
```
Comment (jserv) [inline queue.h:27]: Add a longer description.

File: queue.h [hunk 2]
```diff
@@ -22,35 +22,50 @@ typedef struct {
 
 /* Operations on queue */
 
-/* Create empty queue.
- * Return NULL if could not allocate space.
+/**
+ * q_new() - attempt to create empty queue
```
Comment (jserv) [inline queue.h:26]: Remove the use of "attempt to", which is not necessary.

File: queue.h [hunk 3]
```diff
@@ -60,62 +75,92 @@ bool q_insert_tail(struct list_head *head, char *s);
...
  * https://english.stackexchange.com/questions/52508/difference-between-delete-and-remove
+ *
+ * Return: the pointer to element, %NULL if queue is NULL or empty.
  */
 element_t *q_remove_head(struct list_head *head, char *sp, size_t bufsize);
 
-/* Attempt to remove element from tail of queue.
- * Other attribute is as same as q_remove_head.
+/**
+ * q_remove_head() - attempt to remove element from tail of queue
```
Comment (jserv) [inline queue.h:84]: No "attempt".

File: queue.h [hunk 4]
```diff
@@ -60,62 +75,92 @@ bool q_insert_tail(struct list_head *head, char *s);
...
  */
 element_t *q_remove_tail(struct list_head *head, char *sp, size_t bufsize);
 
-/* Attempt to release element */
-void q_release_element(element_t *e);
-
-/* Return number of elements in queue.
- * Return 0 if q is NULL or empty
+/**
+ * q_release_element() - attempt to release element
```
Comment (jserv) [inline queue.h:94]: Ditto.

Comment (jserv) [reply queue.h:94]: You should emphasize on the internal use.

File: queue.h [hunk 5]
```diff
@@ -60,62 +75,92 @@ bool q_insert_tail(struct list_head *head, char *s);
  *
  * Referenceerence:
```
Comment (jserv) [inline queue.h:76]: Fix typo. Should be "Reference"

**Review by jserv — CHANGES_REQUESTED** [review]
Annotate the members in `element_t` as well.

File: queue.h [hunk 6]
```diff
@@ -22,35 +22,50 @@ typedef struct {
...
+ * q_new() - create an empty queue whose next and prev pointer point to itself
+ *
+ * Return: NULL for allocation failed
  */
 struct list_head *q_new();
 
-/* Free ALL storage used by queue.
- * No effect if q is NULL
+/**
+ * q_free() - attempt to free all storage used by queue, no effect if header is
```
Comment (jserv) [inline queue.h:33]: Ditto. No "attempt"

File: queue.h [hunk 7]
```diff
@@ -60,62 +75,92 @@ bool q_insert_tail(struct list_head *head, char *s);
...
-/* Attempt to release element */
-void q_release_element(element_t *e);
-
-/* Return number of elements in queue.
- * Return 0 if q is NULL or empty
+/**
+ * q_release_element() - attempt to release element
+ * @e: element would be released
+ */
+void static inline q_release_element(element_t *e)
```
Comment (jserv) [inline queue.h:97]: The prototype should be
```c
static inline void q_release_element(element_t *e)
```

File: queue.h [hunk 8]
```diff
@@ -60,62 +75,92 @@ bool q_insert_tail(struct list_head *head, char *s);
...
 void q_reverse(struct list_head *head);
 
-/* Sort elements of queue in ascending order
- * No effect if q is NULL or empty. In addition, if q has only one
- * element, do nothing.
+/**
+ * q_sort() - attempt to sort elements of queue in ascending order
+ * @head: header of queue
+ *
+ * No effect if q is NULL or empty. In addition, if q has only one element, do
```
Comment (jserv) [inline queue.h:162]: `q` is ambiguous. The parameter is one instance of `struct list_head`.

File: queue.h (hunk 4 above)
Comment (eecheng87) [reply queue.h:94]: Should I remove all "attempt" in queue.h?

Comment (jserv) [reply queue.h:94]: Exactly. No "attempt to" in queue.[ch].

File: queue.h [hunk 9]
```diff
@@ -22,35 +22,50 @@ typedef struct {
...
+ *
+ * Return: true for success, false for allocation failed or queue is NULL
  */
 bool q_insert_head(struct list_head *head, char *s);
 
-/* Attempt to insert element at tail of queue.
- * Return true if successful.
- * Return false if q is NULL or could not allocate space.
+/**
+ * q_insert_tail() - attempt to insert element at tail of queue
```
Comment (jserv) [inline queue.h:52]: Correct the grammar. Should be "insert an element at the tail"

File: queue.h [hunk 10]
```diff
@@ -60,62 +75,92 @@ bool q_insert_tail(struct list_head *head, char *s);
...
+ * @e: element would be released
+ */
+void static inline q_release_element(element_t *e)
+{
+    free(e->value);
+    free(e);
+}
+
+/**
+ * q_size() - attempt to get the size of the queue
```
Comment (jserv) [inline queue.h:104]: Capitalize the summary. That is, "Get the size" instead of "get the size."

File: queue.h [hunk 11]
```diff
@@ -11,111 +11,160 @@
...
-/* Attempt to release element */
-void q_release_element(element_t *e);
-
-/* Return number of elements in queue.
- * Return 0 if q is NULL or empty
+/**
+ * q_release_element() - Release element
+ * @e: element would be released
+ *
+ * This function is only for internal use, it should not be modified.
```
Comment (jserv) [inline queue.h:99]: Change the description to the following:
> This function is intended for internal use only.

File: queue.h [hunk 12]
```diff
@@ -11,111 +11,160 @@
...
+ *
+ * Return: true for success, false for allocation failed or queue is NULL
  */
 bool q_insert_tail(struct list_head *head, char *s);
 
-/* Attempt to remove element from head of queue.
- * Return target element.
- * Return NULL if queue is NULL or empty.
+/**
+ * q_remove_head() - Remove element from head of queue
```
Comment (jserv) [inline queue.h:66]: Should be "the element"

File: queue.h [hunk 13]
```diff
@@ -11,111 +11,160 @@
...
  * https://english.stackexchange.com/questions/52508/difference-between-delete-and-remove
+ *
+ * Return: the pointer to element, %NULL if queue is NULL or empty.
  */
 element_t *q_remove_head(struct list_head *head, char *sp, size_t bufsize);
 
-/* Attempt to remove element from tail of queue.
- * Other attribute is as same as q_remove_head.
+/**
+ * q_remove_tail() - Remove element from tail of queue
```
Comment (jserv) [inline queue.h:86]: Ditto.

File: queue.h [hunk 14]
```diff
@@ -11,111 +11,160 @@
...
+ * q_size() - Get the size of the queue
+ * @head: header of queue
+ *
+ * Return: the number of elements in queue, zero if queue is NULL or empty
  */
 int q_size(struct list_head *head);
 
-/* Delete the middle node in list.
+/**
+ * q_delete_mid() - Delete the middle node in list
```
Comment (jserv) [inline queue.h:116]: Don't confuse the developers who read this document. Always say "queue" if you are referring the functions starting with `q_`.

**Pull request merged**